### ISSUE TITLE
Reproducing the results of fine-tuning XLMR large in the paper

### ISSUE THREAD

**Issue by wangxinyu0922** [VinAIResearch/BERTweet#15]
Hi, I'm Interested in your great work and tried to reproduce you results of fine-tuning XLMR (with my own code). And I got `92.6` in `Ritter11`,  `93.4` in `ARK`, `95.0` in `TB-v2`. I find that the results of `ARK` is lower than the reported results in the paper. In the paper, you applied "soft" and "hard" strategy to the dataset while I did nothing. Therefore, I think the reason is possibly the data- preprocessing, am I right?

**Comment by datquocnguyen** [comment]
Yes, probably it is. You might try a soft normalization to see whether there is any difference.
Please also note that ARK result reported in our paper is averaged from 5 different runs. So it might not be surprising that you would obtain small differences in your obtained results compared to the reported ones in our paper.

**Comment by wangxinyu0922** [comment]
I use the preprocessing script for the dataset but I still get a low score on `ARK` (`1327` train, `500` dev and `547` test). I follow the training hyper-parameter in the paper and only got `93.2` with xlmr-large finetuning. By the way, I'm trying using **Bertweet** with transformers but get wrong tokenization (I download the model manually from [transformers](https://huggingface.co/vinai/bertweet-base#) and I failed to use AutoTokenizer to load the vocab.txt):
```
>>> tokenizer = BertTokenizer.from_pretrained("~/.flair/embeddings/bertweet-base/")
Special tokens have been added in the vocabulary, make sure the associated word emebedding are fine-tuned or trained.
>>> line = "SC has first two presumptive cases of coronavirus , DHEC confirms HTTPURL via @USER :cry:"
>>> tokenizer.encode(line)
[63998, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 63996]
>>>
```
The tokenization produce None in the ids. Do you have any idea?

**Comment by datquocnguyen** [comment]
Bertweet was just merged to huggingface's transformers yesterday (https://github.com/huggingface/transformers/pull/6129). You should reinstall `transformers` from the latest source to be able to use Bertweet with AutoTokenizer (and also please remove the `transformers` folder from `~/.cache/torch`). Bertweet has its own tokenizer called `BertweetTokenizer`, not `BertTokenizer`, i.e. you cannot load its tokenizer by using `BertTokenizer`.

    import torch
    from transformers import AutoModel, AutoTokenizer 

    bertweet = AutoModel.from_pretrained("vinai/bertweet-base")
    tokenizer = AutoTokenizer.from_pretrained("vinai/bertweet-base")
    # tokenizer = BertweetTokenizer.from_pretrained("vinai/bertweet-base")

**Comment by datquocnguyen** [comment]
[ARK.zip](https://github.com/VinAIResearch/BERTweet/files/5249468/ARK.zip)
@wangxinyu0922 This is the ARK corpus we use for "soft" evaluation.

**Comment by wangxinyu0922** [comment]
> 
> 
> [ARK.zip](https://github.com/VinAIResearch/BERTweet/files/5249468/ARK.zip)
> @wangxinyu0922 This is the ARK corpus we use for "soft" evaluation.

I tried this dataset with XLM-R large fine-tuning but only get `93.1` on average. Do you have any idea?

**Comment by datquocnguyen** [comment]
No, I dont. I believe in our obtained results, but I do not know what's going on your side.
You might try early stopping as well as evaluate after e.g. every 100 training steps. 
   
    MAX_LENGTH=128
    NUM_EPOCHS=30
    BATCH_SIZE=32
    SAVE_STEPS=100
    LEARNING_RATE=1e-5
